components/tools/OmeroJava/test/integration/chown/PermissionsTest.java

```diff
@@ -238,76 +316,186 @@ public void testChownAnnotationPrivate(boolean isDataOwner, boolean isAdmin, boo
 
         /* note which objects were used to annotate an image */
 
-        final List<IObject> imageAnnotations;
+        final List<IObject> annotations;
+        final List<IObject> annotationsSinglyLinked;
         final List<ImageAnnotationLink> tagLinksOnOtherImage = new ArrayList<ImageAnnotationLink>();
-
-        /* import and annotate an image */
+        final List<ImageAnnotationLink> fileAnnLinksOnOtherImage = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> mapAnnLinksOnOtherImage = new ArrayList<ImageAnnotationLink>();
+        
+        /* import and annotate an image with two sets of annotations */
 
         init(importer);
         final Image image = (Image) iUpdate.saveAndReturnObject(mmFactory.createImage()).proxy();
         final long imageId = image.getId().getValue();
         testImages.add(imageId);
-        imageAnnotations = annotateImage(image);
+        annotations = annotateImage(image);
+        annotationsSinglyLinked = annotateImage(image);
 
-        /* tag another image with the tags from the first image */
+        /* Link Tag, FileAnnotation and MapAnnotation from "annotations" to a second image.
+         * Note that ALL of both "annotations" and "annotationsSinglyLinked" are already linked to the first image.
+         * NONE of the "annotationsSinglyLinked" will be linked to the second image.*/
 
         final Image otherImage = (Image) iUpdate.saveAndReturnObject(mmFactory.createImage()).proxy();
         testImages.add(otherImage.getId().getValue());
-        for (final IObject annotation : imageAnnotations) {
+        for (final IObject annotation : annotations) {
             if (annotation instanceof TagAnnotation) {
                 final ImageAnnotationLink link = (ImageAnnotationLink) annotateImage(otherImage, (TagAnnotation) annotation);
                 tagLinksOnOtherImage.add((ImageAnnotationLink) link.proxy());
             }
+            else if (annotation instanceof FileAnnotation) {
+                final ImageAnnotationLink linkf = (ImageAnnotationLink) annotateImage(otherImage, (FileAnnotation) annotation);
+                fileAnnLinksOnOtherImage.add((ImageAnnotationLink) linkf.proxy());
+            }
+            else if (annotation instanceof MapAnnotation) {
+                final ImageAnnotationLink linkf = (ImageAnnotationLink) annotateImage(otherImage, (MapAnnotation) annotation);
```

Comment:
I can't help but wonder if the `f` of `linkf` stands for the `FileAnnotation` of the above. :smiley:

Might be able to call them all `link` if they're local to the `{ ... }` enclosing block.